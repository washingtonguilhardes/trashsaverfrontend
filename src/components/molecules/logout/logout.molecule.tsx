import * as React from 'react';

import Button, { ButtonProps } from '@mui/material/Button';

interface LogoutProps extends ButtonProps {
  username: string;
  avatar?: string;
  onClickLogout(): void;
}

export function Logout(props: LogoutProps) {
  const { username, onClickLogout, ...others } = props;

  return (
    <Button
      {...others}
      aria-label={username}
      aria-expanded={open ? 'true' : undefined}
      onClick={onClickLogout}
      sx={{ minWidth: 100 }}
      color="inherit"
    >
      Sign Out
    </Button>
  );
}
