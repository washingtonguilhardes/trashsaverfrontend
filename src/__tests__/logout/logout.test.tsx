import { Logout } from '@src/components/molecules/logout';
import { fireEvent, render, screen, waitFor } from '@testing-library/react';

describe('<Logout />', () => {
  test('should render', async () => {
    const onClickLogout = jest.fn();
    render(
      <Logout
        onClickLogout={onClickLogout}
        username="username"
        avatar=""
        data-testid="logout-button"
      />
    );
    const element = screen.getByTestId('logout-button');
    expect(element).toMatchSnapshot();
    fireEvent.click(element);
    await waitFor(() => expect(onClickLogout).toBeCalled());
  });
});
