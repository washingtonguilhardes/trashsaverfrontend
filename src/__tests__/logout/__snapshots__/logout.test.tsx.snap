// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Logout /> should render 1`] = `
<button
  aria-expanded="true"
  aria-label="username"
  avatar=""
  class="MuiButton-root MuiButton-text MuiButton-textInherit MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorInherit MuiButtonBase-root css-1f9myti-MuiButtonBase-root-MuiButton-root"
  data-testid="logout-button"
  tabindex="0"
  type="button"
>
  Sign Out
  <span
    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
  />
</button>
`;
